src/openms/source/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.cpp

```diff
@@ -157,6 +161,171 @@ namespace OpenMS
       }
     }
   }
+  
+  void MRMTransitionGroupPicker::calculatePeakApexInt_(const MSChromatogram& chromatogram,
+    double best_left, double best_right, 
+    ConvexHull2D::PointArrayType & hull_points,
+    double & intensity_sum, 
+    double & rt_sum,
+    double & peak_apex_int,
+    double & peak_apex)
+  {  
+    intensity_sum = 0.0;
+    rt_sum = 0.0;
+    peak_apex_int = -1;
+    double peak_apex_dist = std::fabs(chromatogram.begin()->getMZ() - peak_apex);
+    // FEATURE : use RTBegin / MZBegin -> for this we need to know whether the template param is a real chromatogram or a spectrum!
+    for (MSChromatogram::const_iterator it = chromatogram.begin(); it != chromatogram.end(); it++)
+    {
+      if (it->getMZ() > best_left && it->getMZ() < best_right)
+      {
+        DPosition<2> p;
+        p[0] = it->getMZ();
+        p[1] = it->getIntensity();
+        hull_points.push_back(p);
+        if (std::fabs(it->getMZ() - peak_apex) <= peak_apex_dist)
+        {
+          peak_apex_int = p[1];
+          peak_apex_dist = std::fabs(it->getMZ() - peak_apex);
+        }
+        rt_sum += it->getMZ();
+        intensity_sum += it->getIntensity();
+      }
+    }
+  }
+  
+  void MRMTransitionGroupPicker::calculatePeakShapeMetrics_(const MSChromatogram& chromatogram, 
+    double best_left, double best_right, 
+    double peak_height, double peak_apex, double avg_noise_level,
+    double & width_at_5,
+    double & width_at_10,
+    double & width_at_50,
+    double & start_time_at_10,
+    double & start_time_at_5,
+    double & end_time_at_10,
+    double & end_time_at_5,
+    double & total_width,
+    double & tailing_factor,
+    double & asymmetry_factor,
+    double & baseline_delta_2_height,
+    double & slope_of_baseline,
+    int & points_across_baseline,
+    int & points_across_half_height)
+  {
+    points_across_baseline = 0;
+    double startIntensity, endIntensity;
+    double intensity, intensity_prev, retention_time, retention_time_prev;
+    double start_time_at_50, end_time_at_50;
+    double base, height, height_5, height_10, height_50;
+    
+    // pass 1
+    for (MSChromatogram::const_iterator it = chromatogram.begin(); it != chromatogram.end(); it++)
+    {
+      MSChromatogram::const_iterator it_prev = it;
+      it_prev--; //previous point
+      intensity = std::max(it->getIntensity()-avg_noise_level,0.0); //background-subtracted intensity
+      intensity_prev = std::max(it_prev->getIntensity()-avg_noise_level,0.0); //background-subtracted intensity of the previous point
+      retention_time = it->getMZ();
```

Comment:
getRT()? also below

Reply:
I don't think there is a getRT operator.  At least from what I can see from the documentation.

Reply:
https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/KERNEL/ChromatogramPeak.h#L111